src/stepfunctions/steps/states.py

```diff
@@ -218,9 +218,21 @@ def next(self, next_step):
         Returns:
             State or Chain: Next state or chain that will be transitioned to.
         """
-        if self.type in ('Choice', 'Succeed', 'Fail'):
+        if self.type in ('Succeed', 'Fail'):
             raise ValueError('Unexpected State instance `{step}`, State type `{state_type}` does not support method `next`.'.format(step=next_step, state_type=self.type))
 
+        # By design, choice states do not have the Next field. Setting default to make it chainable.
```

Comment:
explain the `why` rather than the `what` so that future contributors can gain insight.

there's some great info in the [gist](https://gist.github.com/wong-a/e0ef01e4cb717ee277ed90d8eda8dbb7) that @wong-a created that could be paraphrased. Might also be helpful to link to the [language spec](https://states-language.net/spec.html#choice-state)

Reply:
Good point! 
Done!